iroh-base/src/ticket.rs

```diff
@@ -86,6 +86,28 @@ pub enum ParseError {
     Verify { message: &'static str },
 }
 
+impl ParseError {
+    /// Returns a [`ParseError`] that indicates the given ticket has the wrong
+    /// prefix.
+    ///
+    /// Indicate the expected prefix.
+    pub fn wrong_prefix(expected: &'static str) -> Self {
+        KindSnafu { expected }.build()
+    }
+
+    /// Return a `ParseError` variant that indicates verification of the
+    /// deserialized bytes failed.
+    pub fn verification_failed(message: &'static str) -> Self {
+        VerifySnafu { message }.build()
+    }
+}
+
+impl From<postcard::Error> for ParseError {
+    fn from(source: postcard::Error) -> Self {
+        PostcardSnafu.into_error(source)
```

Comment:
you don't need to do this manually: https://docs.rs/snafu/latest/snafu/derive.Snafu.html#disabling-the-context-selector

Reply:
yeah probably